lib/install/deps.js

```diff
@@ -671,6 +671,7 @@ var findRequirement = exports.findRequirement = function (tree, name, requested,
     return null
   }
   if (tree.isTop) return null
+  if (!preserveSymlinks() && /^[.][.][\\/]/.test(path.relative(tree.parent.realpath, tree.realpath))) return null
```

Comment:
Apparently something about our previously broken `isLink`/`isInLink` handling was papering over this bug. `findRequirement` needs to not count things above links-outside-our-project as viable dependencies.